File: backend/src/main/java/org/example/backend/recipe/model/IngredientModel.java
package org.example.backend.recipe.model;

import com.mongodb.lang.Nullable;
import lombok.With;

import java.util.List;

@With
public record IngredientModel(
        String id,
        String name,
        List<NutritionValuesPerUnitModel> nutritionValuesPerUnit,
        @Nullable String[] allergens
) {
}

File: frontend/src/models/recipe/IngredientModel.ts
import type { NutritionValuesPerUnitModel } from "./NutritionValuesPerUnitModel.ts";

export type IngredientModel = {
  id: string;
  name: string;
  nutritionValuesPerUnit: NutritionValuesPerUnitModel[];
  allergens: string[] | null;
};
